Sources/QueuesFluentDriver/FluentQueue.swift

```diff
@@ -1,103 +1,115 @@
 
-public struct FluentQueue {
+/// [SE-0418] negates the need for this ugliness, but until then, to silence the Sendability warnings for the keypaths,
+/// we mark keypaths with the relevant root type unchecked. As this is a retroactive conformance for a type not owned
+/// by this module, per [SE-0364] we need to explicitly mark it so when the attribute is available.
+///
+/// [SE-0418]: https://github.com/apple/swift-evolution/blob/main/proposals/0418-inferring-sendable-for-methods.md
+/// [SE-0364]: https://github.com/apple/swift-evolution/blob/main/proposals/0364-retroactive-conformance-warning.md
+#if compiler(<5.10) || !$InferSendableFromCaptures
+#if compiler(>=5.10) && $RetroactiveAttribute
+extension KeyPath: @retroactive @unchecked Sendable where Root == JobDataModel {}
+#else
+extension KeyPath: @unchecked Sendable where Root == JobDataModel {}
+#endif
+#endif
+
+public struct FluentQueue: Queue, Sendable {
     public let context: QueueContext
-    let db: Database
-    let dbType: QueuesFluentDbType
-    let useSoftDeletes: Bool
-}
 
-extension FluentQueue: Queue {
+    let db: any Database
+    let sqlDb: any SQLDatabase
+
     public func get(_ id: JobIdentifier) -> EventLoopFuture<JobData> {
-        return db.query(JobModel.self)
-            .filter(\.$id == id.string)
+        self.sqlDb.select()
+            .column(JobModel.sqlColumn(\.$data.$payload))
+            .column(JobModel.sqlColumn(\.$data.$maxRetryCount))
+            .column(JobModel.sqlColumn(\.$data.$jobName))
+            .column(JobModel.sqlColumn(\.$data.$delayUntil))
+            .column(JobModel.sqlColumn(\.$data.$queuedAt))
+            .column(JobModel.sqlColumn(\.$data.$attempts))
+            .from(JobModel.sqlTable)
+            .where(JobModel.sqlColumn(\.$id), .equal, SQLBind(id.string))
             .first()
             .unwrap(or: QueuesFluentError.missingJob(id))
-            .flatMapThrowing { job in
-                return JobData(
-                    payload: Array(job.data.payload),
-                    maxRetryCount: job.data.maxRetryCount,
-                    jobName: job.data.jobName,
-                    delayUntil: job.data.delayUntil,
-                    queuedAt: job.data.queuedAt,
-                    attempts: job.data.attempts ?? 0
+            .flatMapThrowing { row in
+                try row.decode(
+                    model: JobData.self,
+                    prefix: "\(JobModel().$data.key)_",
+                    keyDecodingStrategy: .convertFromSnakeCase
                 )
             }
     }
     
     public func set(_ id: JobIdentifier, to jobStorage: JobData) -> EventLoopFuture<Void> {
         let jobModel = JobModel(id: id, queue: queueName.string, jobData: JobDataModel(jobData: jobStorage))
-        // If the job must run at a later time, ensure it won't be picked earlier since
-        // we sort pending jobs by date when querying
-        jobModel.runAtOrAfter = jobStorage.delayUntil ?? Date()
```

Comment:
Do we not need to worry about this anymore?

Reply:
I believe not - see https://github.com/vapor-community/vapor-queues-fluent-driver/pull/1#discussion_r1471850513